### ISSUE TITLE
GET v2/wallets does not list wallets from oldest to newest

### ISSUE THREAD

**Issue by piotr-iohk** [cardano-foundation/cardano-wallet#250]
| Release | Operating System       | Cause |
| ---     | ---                    | ---   |
| next    | Linux| Code or Human or Unknown |

# Context
 
https://input-output-hk.github.io/cardano-wallet/api/#operation/listWallets says `Return a list of known wallets, ordered from oldest to newest.` which currently it not the case.

# Steps to Reproduce

1. Add wallets "1", "2" and "3" in the following order:

```
curl  -vX POST http://localhost:8090/v2/wallets \
  -H "Content-Type: application/json; charset=utf-8" \
  -d '{
"name": "1",
"mnemonic_sentence": ["network", "empty", "cause", "mean", "expire", "private",
        "finger", "accident", "session", "problem", "absurd", "banner", "stage",
        "void", "what"],
"passphrase": "Secure Passphrase",
"address_pool_gap": 20
}' --http1.1  |  jq


curl  -vX POST http://localhost:8090/v2/wallets \
  -H "Content-Type: application/json; charset=utf-8" \
  -d '{
"name": "2",
"mnemonic_sentence": ["whisper", "control", "diary", "solid", "cattle", "salmon",
        "whale", "slender", "spread", "ice", "shock", "solve", "panel",
        "caution", "upon", "scatter", "broken", "tonight"],
"passphrase": "Secure Passphrase",
"address_pool_gap": 20
}' --http1.1  |  jq


curl  -vX POST http://localhost:8090/v2/wallets \
  -H "Content-Type: application/json; charset=utf-8" \
  -d '{
"name": "3",
"mnemonic_sentence": ["click", "puzzle", "athlete", "morning", "fold", "retreat",
        "across", "timber", "essay", "drill", "finger", "erase", "galaxy",
        "spoon", "swift", "eye", "awesome", "shrimp", "depend", "zebra", "token"],
"passphrase": "Secure Passphrase",
"address_pool_gap": 20
}' --http1.1  |  jq
```

2. List added wallets:

`curl  -vX GET http://localhost:8090/v2/wallets |  jq`

## Expected behavior

Should be listed  (by wallet name):
1
2
3

## Actual behavior
 Wallets are listed in the following order (by wallet name):
3
1
2

<A clear and concise description of what you observe instead. If applicable add screenshots to help explain your problem.>

---

# Resolution Plan

Required:
- [x] Add `creationTime` field to `Cardano.Wallet.Primitive.Types.WalletMetadata`
- [x] Incrementally recompile the code, correcting errors until the code compiles successfully.
- [x] Add appropriate API tests to demonstrate that wallets are listed in the correct order.
- [x] Make the tests pass by performing sorting in the appropriate place.

Optional:
- [ ] Add `creationTime` field to `Cardano.Wallet.Api.Types.ApiWallet`

# PR

| Number   | Base      |
| ---      | ---       |
| #344        | `develop` |

# QA

**Comment by jonathanknowles** [comment]
@KtorZ Do you know if there's an operation or interface that makes it possible to determine the "creation time" of a wallet? (I had a brief look through the code, but couldn't spot anything relevant.) If not, how we would go about determining this info? (And where should it be stored?)

**Comment by KtorZ** [comment]
We cannot do this at the moment indeed. So we should have some sort of timestamp somewhere. What makes most sense is probably to have that within the wallet metadata and store it upon creation.

**Comment by jonathanknowles** [comment]
Fix available: #344

**Comment by piotr-iohk** [comment]
fixed :+1:
